test/wallet-account.test.js

```diff
@@ -185,3 +202,69 @@ it('requests transaction signing automatically when there is no local key', asyn
     expect(transactions[0].actions[0].transfer.deposit.toString()).toEqual('1');
     expect(Buffer.from(transactions[0].publicKey.data)).toEqual(Buffer.from(keyPair.publicKey.data));
 });
+
+it('requests transaction signing automatically when function call has attached deposit', async () => {
+    let localKeyPair = nearApi.KeyPair.fromRandom('ed25519');
+    let walletKeyPair = nearApi.KeyPair.fromRandom('ed25519');
+    setupWalletConnectionForSigning({
+        allKeys: [ walletKeyPair.publicKey.toString() ],
+        accountAccessKeys: [{
+            access_key: {
+                nonce: 1,
+                permission: {
+                    FunctionCall: {
+                        allowance: '1000000000',
+                        receiver_id: 'receiver.near',
+                        method_names: []
+                    }
+                }
+            },
+            public_key: localKeyPair.publicKey.toString()
+        }, {
+            access_key: {
+                nonce: 1,
+                permission: 'FullAccess'
+            },
+            public_key: walletKeyPair.publicKey.toString()
+        }]
+    });
+    keyStore.setKey('networkId', 'signer.near', localKeyPair);
+
+    try {
+        await walletConnection.account().signAndSendTransaction('receiver.near', [
+            nearApi.transactions.functionCall('someMethod', new Uint8Array(), new BN('1'), new BN('1'))
+        ]);
+        fail('expected to throw');
+    } catch (e) {
+        expect(e.message).toEqual('Failed to redirect to sign transaction');
+    }
```

Comment:
I don't mind the `try / catch`, but it may clean it up a bit to use `expect(() => { ... }).toThrow('Failed to redirect...')` 

See https://jestjs.io/docs/en/expect#tothrowerror

Reply:
I don't like this pattern as it is limits you on what you can assert about error (e.g. once we have codes in every error – looks like it won't be convenient to verify both message and code).